docs/fluttertodostutorial.md

```diff
@@ -1,483 +1,1488 @@
+
 # Flutter Todos Tutorial
 
 ![advanced](https://img.shields.io/badge/level-advanced-red.svg)
 
 > In the following tutorial, we're going to build a Todos App in Flutter using the Bloc library.
 
 ![demo](./assets/gifs/flutter_todos.gif)
+> ZZTODO: Update gif with the movie in the issue tracker.
 
 ## Key Topics
 
 - Observe state changes with [BlocObserver](/coreconcepts?id=blocobserver).
 - [BlocProvider](/flutterbloccoreconcepts?id=blocprovider), Flutter widget which provides a bloc to its children.
 - [BlocBuilder](/flutterbloccoreconcepts?id=blocbuilder), Flutter widget that handles building the widget in response to new states.
-- Using Bloc instead of Cubit. [What's the difference?](/coreconcepts?id=cubit-vs-bloc)
+- [BlocListener](/flutterbloccoreconcepts?id=bloclistener), a Flutter widget which invokes the listener code in response to state changes in the bloc.
+- Using 3 Blocs and  1 Cubit. [What's the difference?](/coreconcepts?id=cubit-vs-bloc)
+- Uses a [Repository and API layer](architecture/repository) separately for the data.
+- [RespositoryProvider](/flutterbloccoreconcepts?id=respositoryprovider), a flutter widget to provide a repository to its children.
 - Prevent unnecessary rebuilds with [Equatable](/faqs?id=when-to-use-equatable).
-- [MultiBlocProvider](/flutterbloccoreconcepts?id=multiblocprovider), a Flutter widget that merges multiple BlocProvider widgets into one.
+- [MultiBlocListener](/flutterbloccoreconcepts?id=multibloclistener), a Flutter widget that merges multiple BlocListener widgets into one.
+
 
 ## Setup
 
 We'll start off by creating a brand new Flutter project
 
-[script](_snippets/flutter_todos_tutorial/flutter_create.sh.md ':include')
+```shell
+flutter create flutter_login
+```
 
 We can then replace the contents of `pubspec.yaml` with
 
 [pubspec.yaml](_snippets/flutter_todos_tutorial/pubspec.yaml.md ':include')
 
 and then install all of the dependencies
 
-[script](_snippets/flutter_todos_tutorial/flutter_packages_get.sh.md ':include')
+```shell
+flutter pacakages get
+```
+
+[comment]: <> (> ZTODO: Is below still true? I think so, but I don't know the history of this.)
+[comment]: <> (> **Note:** We're overriding some dependencies because we're going to be reusing them from [Brian Egan's Flutter Architecture Samples]&#40;https://github.com/brianegan/flutter_architecture_samples&#41;.)
+
+## Directory Structure
+
+Because this is a long tutorial, presenting the directory structure is helpful.
+
+> Alternatively, explore the files on github or clone them locally and open in your IDE. The full source for this example can be found [here](https://github.com/felangel/Bloc/tree/master/examples/flutter_todos).
+
+We will be using 3 blocs, 1 cubit, and 3 packages.
```

Comment:
I think we should explain this in sections, @chonghorizons I can share with you the outline I had for this tutorial so we can just explain the components future-proofing the quantities we’re using right now in case we make the tutorial more complex, we just need to add a new section and not changing these numbers, wdyt?

Reply:
That would be okay, but I think the numbers are helpful for newcomers reading the docs. It's also pointing out to look at the directory structure to see this.

It could be more clear by breaking it out as a

> TIP: Look at the directory structure. By using a good naming convention and directory structure, we can easily see: We will be using 3 blocs, 1 cubit, and 3 packages.

Sorry, I can't edit directly since I'm not a collaborator.